client/landing/stepper/hooks/use-record-signup-complete.ts

```diff
@@ -3,68 +3,81 @@ import { useSelect } from '@wordpress/data';
 import { useCallback } from 'react';
 import { USER_STORE, ONBOARD_STORE } from 'calypso/landing/stepper/stores';
 import { SIGNUP_DOMAIN_ORIGIN, recordSignupComplete } from 'calypso/lib/analytics/signup';
+import { useSelector } from 'calypso/state';
+import isUserRegistrationDaysWithinRange from 'calypso/state/selectors/is-user-registration-days-within-range';
 import { useSite } from './use-site';
 import type { UserSelect, OnboardSelect } from '@automattic/data-stores';
 
 export const useRecordSignupComplete = ( flow: string | null ) => {
 	const site = useSite();
```

Comment:
@escapemanuele is this where the issue stems? It looks like it grabs site ID/slug from URL params, which would obviously be `undefined` unless carried along?

Isn't there a store/state slice that gets populated with site ID when one is created? We should probably look at create-site step and do that if not?

Reply:
What I ended up doing is passing the data from the processing step directly when calling the useRecordSignupComplete function [here](https://github.com/Automattic/wp-calypso/pull/94706/files#diff-b7ff3243075883b8ecc07a8d6f6e9c94a960ec6d179ac29676a74d6270a6fe1aR107). At that point we have all the data needed. Do you think that should be changed?

Reply:
Cool. Sounds good (you obviously know more about these concepts)

> Do you think that should be changed?

I'm not sure. It might not be a bad idea to populate the onboard store with it once a site is created (or pending). It would make it available in anything that follows and needs it.

I will comment up above on something related if you can also take a look pls